RosettaJdbi3/src/main/java/com/hubspot/rosetta/jdbi3/RosettaRowMapperFactory.java

```diff
@@ -0,0 +1,52 @@
+package com.hubspot.rosetta.jdbi3;
+
+import java.lang.reflect.Type;
+import java.util.Optional;
+
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.generic.GenericTypes;
+import org.jdbi.v3.core.mapper.BuiltInMapperFactory;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.mapper.RowMapperFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hubspot.rosetta.RosettaMapper;
+import com.hubspot.rosetta.util.SqlTableNameExtractor;
+
+public class RosettaRowMapperFactory implements RowMapperFactory {
+  private static final BuiltInMapperFactory BUILT_INS = new BuiltInMapperFactory();
+
+  @Override
+  public Optional<RowMapper<?>> build(Type type, ConfigRegistry config) {
+    if (accepts(type, config)) {
+      return Optional.of((rs, ctx) -> {
+        ObjectMapper objectMapper = RosettaObjectMapperOverride.resolve(ctx);
+
+        String tableName = SqlTableNameExtractor.extractTableName(ctx.getParsedSql().getSql());
+        final RosettaMapper mapper = new RosettaMapper(type, objectMapper, tableName);
+
+        return mapper.mapRow(rs);
+      });
+    } else {
+      return Optional.empty();
+    }
+  }
```

Comment:
In Jdbi3, we've moved away from mapper factories eagerly trying to map every type. It tends to cast too wide a net and try to map things you didn't intend.

I would recommend factory methods to create mappers for specific types. See e.g. `BeanMapper.factory(type)`

Reply:
I think Rosetta is somewhat intentionally a bad neighbor in this regard. One of the main benefits of the library is that it offers a set-it-and-forget-it approach to mapping; you can register it globally and just let it do its thing. If there's an exceptional case that is better handled with a custom mapper, you can always register a custom mapper with `@RegisterRowMapper` (basically using Rosetta as the default with the ability to opt-out vs. having no default and needing to manually opt-in everything). Does this seem reasonable?